src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -556,17 +527,17 @@ public String getApiMethodName(Method method, VisibilityConfig visibility) {
   }
 
   /** The name of the async surface method which can call the given API method. */
-  public String getAsyncApiMethodName(Method method) {
-    return publicMethodName(Name.upperCamel(method.getSimpleName()).join("async"));
+  public String getAsyncApiMethodName(Method method, VisibilityConfig visibility) {
+    return visibility.methodName(this, Name.upperCamel(method.getSimpleName()).join("async"));
   }
 
   /** The name of the example for the method. */
   public String getApiMethodExampleName(Interface interfaze, Method method) {
     return getApiMethodName(method, VisibilityConfig.PUBLIC);
   }
 
-  public String getAsyncApiMethodExampleName(Method method) {
-    return getNotImplementedString("SurfaceNamer.getAsyncApiMethodExampleName");
+  public String getAsyncApiMethodExampleName(Interface interfaze, Method method) {
```

Comment:
The thought was that since `getApiMethodExampleName` in Go needs Interface, this should be consistent in case it needs to be overridden.